presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -182,28 +220,154 @@ public void setBinaryStream(int parameterIndex, InputStream x, int length)
     public void clearParameters()
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "clearParameters");
+        checkOpen();
+        parameters.clear();
     }
 
     @Override
     public void setObject(int parameterIndex, Object x, int targetSqlType)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setObject");
+        checkOpen();
+        if (x == null) {
+            setNull(parameterIndex, targetSqlType);
+            return;
+        }
+        switch (targetSqlType) {
+            case Types.NULL:
+                setNull(parameterIndex, targetSqlType);
+                break;
+            case Types.TINYINT:
+                setByte(parameterIndex, (x instanceof Number) ? ((Number) x).byteValue() : Byte.valueOf(String.valueOf(x)));
+                break;
+            case Types.SMALLINT:
+                setShort(parameterIndex, (x instanceof Number) ? ((Number) x).shortValue() : Short.valueOf(String.valueOf(x)));
```

Comment:
Do we need this `String.valueOf(` elasticity? It means `setObject(1, o, Types.SMALLINT)` is going to accept many weird `o` types, just because they `toString` to something numerical. 

 http://download.oracle.com/otndocs/jcp/jdbc-4_2-mrel2-spec/index.html > table B-5 "Conversions by setObject and setNull from Java Object Types to JDBC Types" documents the conversions that need to be supported by the JDBC impl. 
For example, for Types.SMALLINT it mandates following types should be accepted:
- numerical
- String
- boolean (this now would fail)
- ... (i didn't grep the table carefully)

Also, if the conversion cannot be done (like `Short.valueOf("aaa")`), `SQLException` should be thrown.

This comment applies to all (or almost all) cases here, incl `Types.VARCHAR` itself.

Reply:
Thanks for the link! I'll update this to be more strict and add additional tests.